metashare/api/jobs.py

```diff
@@ -0,0 +1,178 @@
+import contextlib
+import logging
+import traceback
+
+from asgiref.sync import async_to_sync
+from cumulusci.core.config import ScratchOrgConfig
+from django.utils.text import slugify
+from django.utils.timezone import now
+from django_rq import job
+from github3 import login
+from github3.exceptions import UnprocessableEntity
+
+from .github_context import (
+    extract_owner_and_repo,
+    get_cumulus_prefix,
+    local_github_checkout,
+)
+from .push import push_message_about_instance
+from .sf_run_flow import run_flow
+
+logger = logging.getLogger(__name__)
+
+
+async def report_scratch_org_error(instance, message):
+    from .serializers import ScratchOrgSerializer
+
+    message = {
+        "type": "SCRATCH_ORG_PROVISION_FAILED",
+        "payload": {"message": message, "model": ScratchOrgSerializer(instance).data},
+    }
+    await push_message_about_instance(instance, message)
+
+
+@contextlib.contextmanager
+def report_errors_on(scratch_org):
+    try:
+        yield
+    except Exception as e:
+        async_to_sync(report_scratch_org_error)(scratch_org, str(e))
+        tb = traceback.format_exc()
+        logger.error(tb)
+        scratch_org.delete()
+        raise
+
+
+def create_scratch_org(*, scratch_org, user, repo_url, commit_ish):
+    """
+    Expects to be called in the context of a local github checkout.
+    """
+    gh = login(token=user.gh_token)
+    owner, repo = extract_owner_and_repo(repo_url)
+    repository = gh.repository(owner, repo)
+    branch = repository.branch(commit_ish)
+    commit = branch.commit
+    latest_commit = commit.sha
+    latest_commit_url = commit.html_url
+    latest_commit_at = commit.commit.author.get("date", None)
+
+    org_config = ScratchOrgConfig(
+        {"config_file": "orgs/dev.json", "scratch": True}, "dev"
+    )
+    org_config.create_org()
```

Comment:
Pull out these two lines and use the structure in sf_run_flows, to avoid using sfdx.